stack/__main__.py

```diff
@@ -872,6 +872,15 @@ def find_service_account(kind: str, access_level: str) -> Optional[str]:
             opts=pulumi.resource.ResourceOptions(depends_on=[cloudidentity]),
         )
 
+        # allow all hail service_accounts to access the server-config
+        gcp.secretmanager.SecretIamMember(
```

Comment:
I think that’s problematic, as the secret contains tokens for all projects (and all access levels). That would allow any Hail service account to act as any other.

Reply:
Yep, totally agree. And I'm glad I went back to look more closely at this. This can actually just be addressed by adding a `dataset-to-sample-id` map. I've put this into a secret, and in fact I can update the access_group_cache to use this (instead of the server-config).